tests/31sync/03joined.pl

```diff
@@ -124,3 +124,67 @@
          Future->done(1)
       })
    };
+
+
+test "Newly joined room has correct timeline in incremental sync",
+   requires => [qw( first_api_client can_sync )],
+
+   check => sub {
+      my ( $http ) = @_;
+
+      my ( $user_a, $filter_id_a, $user_b, $filter_id_b, $room_id, $next_b );
+
+      my $filter = {
+         room => {
+            timeline => { types => [ "m.room.message" ], limit => 10 },
+            state    => { types => [] },
+         }
+      };
+
+      Future->needs_all(
+         matrix_register_user_with_filter( $http, $filter )
+            ->on_done( sub { ( $user_a, $filter_id_a ) = @_ } ),
+         matrix_register_user_with_filter( $http, $filter )
+            ->on_done( sub { ( $user_b, $filter_id_b ) = @_ } ),
+      )->then( sub {
+         matrix_create_room( $user_a )->on_done( sub { ( $room_id ) = @_ } );
+      })->then( sub {
+         Future->needs_all( map {
+            matrix_send_room_text_message( $user_a, $room_id, body => "test" );
+         } 0 .. 3 );
+      })->then( sub {
+         matrix_sync( $user_b, filter => $filter_id_b )->on_done( sub {
+            my ( $body ) = @_;
+
+            $next_b = $body->{next_batch};
+         });
+      })->then( sub {
+         Future->needs_all( map {
+            matrix_send_room_text_message( $user_a, $room_id, body => "test" );
+         } 0 .. 3 );
+      })->then( sub {
+         matrix_join_room( $user_b, $room_id );
+      })->then( sub {
+         matrix_sync( $user_b, filter => $filter_id_b, since => $next_b );
+      })->then( sub {
+         my ( $body ) = @_;
+         my $room = $body->{rooms}{joined}{$room_id};
+         my $timeline = $room->{timeline};
+
+         map {
+            $room->{event_map}{$_}{type} eq "m.room.message"
+               or die "Only expected 'm.room.message' events";
+         } @{ $timeline->{events} };
+
+         if( @{ $timeline->{events} } == 6 ) {
+            $timeline->{limited} == JSON::false
+               or die "Timeline doesn't have all the events so should be limited";
+         }
+         else {
+            $timeline->{limited} == JSON::true
```

Comment:
Is there a particular reason to actually compare this to JSON::true (and false above); or just testing for truthiness? If not, then I'd suggest just use the truth of the value, rather than this full comparison.